src/main/kotlin/no/nav/familie/ef/iverksett/infrastruktur/service/KafkaProducerService.kt

```diff
@@ -6,7 +6,7 @@ import org.springframework.stereotype.Service
 @Service
 class KafkaProducerService(private val kafkaTemplate: KafkaTemplate<String, String>) {
 
-    fun send(topic: String, payload: String) {
-        kafkaTemplate.send(topic, payload).get()
+    fun send(topic: String, key: String, payload: String) {
+        kafkaTemplate.send(topic, key, payload).get()
```

Comment:
Er KEY unik? For behandlingsstatistikk så sender vi vel flere ganger, hva skjer når vi sender gang nr 2?

Reply:
Gikk ut ifra at behandlingId er den mest unike variabelen vi har, så brukte den. Det er egentlig bare for å avgjøre hvilken partisjon meldingen skal på. Har bare satt opp bruk av 1 partisjon, da ytelse ikke er noe tema enda. Key er et must dersom man har compact som retention policy, for brukes key til å slettes de eldste versjonene av records med lik key. Dersom delete settes som retention policy blir det dermed ikke et krav. Men har skjønt at det er good practice å sende med key uansett, til tross for at vi hittil bare har 1 partisjon og har delete som retention policy. Setter retention policy til delete da det er default.

Reply:
Ok 👍